models/DataObject/AbstractObject.php

```diff
@@ -516,7 +516,7 @@ public static function getTotalCount($config = [])
      */
     protected static function typeMatch(AbstractObject $object)
     {
-        return in_array(static::class, [Concrete::class, __CLASS__], true) || $object instanceof static;
+        return static::class === __CLASS__ || $object instanceof static;
```

Comment:
This change can break implementations relying on `Concrete::getById()` to load folder objects. Maybe we just trigger deprecation here and change the behavior in Pimcore 11.

Reply:
```suggestion
        if (static::class === Concrete::class && !$object instanceof static) {
            trigger_deprecation(
                'pimcore/pimcore',
                '10.5',
                'Loading none Concrete objects with via the Concrete class will not possible in Pimcore 11'
            );

            return true;
        }

        return static::class === __CLASS__ || $object instanceof static;
```